pkg/master/build.go

```diff
@@ -116,22 +115,22 @@ func (phase *serverPhaseBuild) CreatePackage(in *service.PackageNode) (*service.
 	if _, err := phase.db.GetPackageNodeByName(in.Name); err != database.ErrNoSuchPackage {
 		return nil, errors.New("package already created")
 	}
-
 	log.Printf("Adding package node %s", in.Name)
 	phase.db.AddPackageNode(in)
 	return &service.BuildResponse{Success: true}, nil
 }
 
-func (phase *serverPhaseBuild) CreateProject(in *service.ProjectNode) (*service.BuildResponse, error) {
-	if !in.IsValid() {
+func (phase *serverPhaseBuild) CreateProject(in *service.ProjectMessage) (*service.BuildResponse, error) {
+	if !in.Project.IsValid() {
 		return nil, errors.New("invalid project node")
 	}
-	if _, err := phase.db.GetProjectNode(); err != database.ErrNoProjectNode {
-		return nil, errors.New("project node already created")
+	if !in.Update {
```

Comment:
This is rather useless. I the boolean is set to false you could override every project node.
Also this needs adaptation of the proto file.

Why not just checking for a set uid?